lmdeploy/serve/openai/api_server.py

```diff
@@ -271,15 +425,18 @@ async def generate(request: GenerateRequest, raw_request: Request = None):
     - ignore_eos (bool): indicator for ignoring eos
     """
     if request.session_id == -1:
-        session_id = ip2id(raw_request.client.host)
-        request.session_id = session_id
+        request.session_id = random.randint(10087, 23333)
+
+    async_engine = VariableInterface.async_engine
+    sequence_start = async_engine.steps.get(str(request.session_id), 0) == 0
+    sequence_end = not request.interactive_mode
```

Comment:
If `interactive_mode` is True, `sequence_end` is always False. How can use end a session? 
One way is to add `end` field in `GenerateRequest`. Any idea?

Reply:
There is a function named `end_session` in APIClient.